KAFKA-19152: Add top-level documentation for OAuth flows

kirktrue opened the pull request:
Adds documentation to support the OAuth additions from KIP-768 and
KIP-1139.

The existing documentation is heavily geared toward Kafka's support for
non-production OAuth usage. Since this mode is still supported, it
should not be removed. However, with the addition of the production
OAuth usage, the documentation is less than succinct because it has a
bit of a split personality issue.

kirktrue committed:
Add top-level documentation for OAuth flows

kirktrue committed:
Update security.html

Yunyung commented on docs/security.html:
```diff
@@ -927,8 +963,40 @@ <h3 class="anchor-heading"><a id="security_sasl" class="anchor-link"></a><a href
                 </li>
+                <li><h5 class="anchor-heading"><a id="security_sasl_oauthbearer_production_brokerconfig" class="anchor-link"></a><a href="#security_sasl_oauthbearer_production_brokerconfig">Configuring Production Kafka Brokers</a></h5>
+                    <ol>
+                        <li>Add a suitably modified JAAS file similar to the one below to each Kafka broker's config directory, let's call it kafka_server_jaas.conf for this example:
```
Looks like there's no corresponding "\</li\>"

kirktrue replied:
Fixed. Thanks!

Yunyung commented on docs/security.html:
```diff
@@ -556,8 +556,8 @@ <h3 class="anchor-heading"><a id="security_sasl" class="anchor-link"></a><a href
-                        <a href="#security_sasl_scram_brokerconfig">SCRAM</a> or
-                        <a href="#security_sasl_oauthbearer_brokerconfig">OAUTHBEARER</a> for example broker configurations.</p></li>
+                        <a href="#security_sasl_scram_brokerconfig">SCRAM</a>, or
+                        <a href="#security_sasl_oauthbearer_non_production_brokerconfig">non-production</a>/<a href="#security_sasl_oauthbearer_non_production_brokerconfig">production</a> OAUTHBEARER for example broker configurations.</p></li>
```
```suggestion
                        <a href="#security_sasl_oauthbearer_non_production_brokerconfig">non-production</a>/<a href="#security_sasl_oauthbearer_production_clientconfig">production</a> OAUTHBEARER for example broker configurations.</p></li>
```

kirktrue replied:
Fixed to be `security_sasl_oauthbearer_production_brokerconfig`.

Yunyung commented on docs/security.html:
```diff
@@ -579,8 +579,8 @@ <h3 class="anchor-heading"><a id="security_sasl" class="anchor-link"></a><a href
-                                <a href="#security_sasl_scram_clientconfig">SCRAM</a> or
-                                <a href="#security_sasl_oauthbearer_clientconfig">OAUTHBEARER</a> for example configurations.</p></li>
+                                <a href="#security_sasl_scram_clientconfig">SCRAM</a>, or
+                                <a href="#security_sasl_oauthbearer_non_production_clientconfig">non-production</a>/<a href="#security_sasl_oauthbearer_production_clientconfig">production</a> OAUTHBEARER for example configurations.</p></li>
```
Nit: Corresponding to "for example broker configurations" above.
```suggestion
                                <a href="#security_sasl_oauthbearer_non_production_clientconfig">non-production</a>/<a href="#security_sasl_oauthbearer_production_clientconfig">production</a> OAUTHBEARER for example client configurations.</p></li>
```

kirktrue replied:
Added "client" qualifier.

Yunyung reviewed (commented):
Thanks for the PR. I left some comments.

kirktrue committed:
Tidying up links and tags

Fixed typo in broker configuration link, added qualifier for client configuration link, and properly closed the HTML list.

kirktrue commented:
Thanks @Yunyung.

Ready for another round of review.

cc @lianetm @omkreddy

kirktrue committed:
Merge branch 'trunk' into KAFKA-19152-oauth-docs
